src/main/java/com/voti/pawction/scheduling/AuctionScheduler.java

```
@@ -32,8 +33,12 @@ public void closeExpired() {
                 log.info("[auction-scheduler] closed {} auctions (cutoff {}, grace {}s)",
                         closed, cutoff, props.getGraceSeconds());
             }
-        } catch (Exception e) {
-            log.error("[auction-scheduler] error (cutoff {}, grace {}s)", cutoff, props.getGraceSeconds(), e);
+        } 
+        // Narrow the exception handling from Exception → RuntimeException
+        catch (RuntimeException e) {
+            log.error("[auction-scheduler] runtime error while closing expired auctions " +
+                            "(cutoff {}, grace {}s)",
+                    cutoff, props.getGraceSeconds(), e);
```

Narrowed exception handling from Exception to RuntimeException to avoid swallowing checked exceptions and to make scheduler failures more intentional.